index.js

```diff
@@ -1,37 +1,48 @@
-var fs = require('fs');
+const fs = require('fs');
+const path = require('path');
+
+function addHook(compiler, pluginName, hookName, hook) {
+  if ('hooks' in compiler) {
+    compiler.hooks[hookName].tap(pluginName, hook);
+  } else {
+    compiler.plugin(hookName, hook);
+  }
+}
 
 function WebpackTouch(options) {
-  var options = options || {}
-  this.filename = options.filename;
-  this.delay = options.delay? parseInt(options.delay) : 0;
-  this.ignoreOnError = options.ignoreOnError;
-  if (!this.filename) {
-    throw new Error('Require filename option')
+  if (!options.filename) {
```

Comment:
(This.)options.filename should be after options || {} bellow since options can be undefined, operating options.filename on undefined will trigger error

Reply:
Omg, thanks. Fixed it.